controllers/cluster_controller.go

```diff
@@ -243,6 +244,14 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 	}
 	controlPlaneMachines, machines := splitMachineList(allMachines)
 
+	secrets := &corev1.SecretList{}
+	secretsListOptions := []client.ListOption{
```

Comment:
Secret list has it's own list options because we don't apply the `cluster.x-k8s.io/cluster-name` label to secret resources (maybe we should?).